File: frontend/src/app/endpoints/kupac-endpoints/get-restorani.endpoint.ts
import {Injectable} from "@angular/core";
import {MyBaseEndpoint} from "../MyBaseEndpoint";
import {Observable} from "rxjs";
import {HttpClient} from "@angular/common/http";
import {MojConfig} from "../../moj-config";

@Injectable({providedIn:'any'})

export class GetRestoraniEndpoint implements MyBaseEndpoint<void, RestoraniGetResponse>{

  constructor(public httpClient:HttpClient) {
  }
  akcija(request:void): Observable<RestoraniGetResponse> {
    let url = MojConfig.adresa_servera + `/Kupac-GetRestorani`;

    return this.httpClient.get<RestoraniGetResponse>(url);
  }
}


export interface RestoraniGetResponse {
  restorani: RestoraniGetResponseRestoran[]
  brojRestorana: number
}

export interface RestoraniGetResponseRestoran{
  id: number
  naziv: string
  radnoVrijemeOd: string
  radnoVrijemeDo: string
  opis: string
  slikaRestorana: any
  isObrisan: boolean
  lokacija: string
  imePrezimeVlasnika: string
}

File: http/http/Endpoint/Kupac/GetRestorani/GetRestoraniEndpoint.cs
using http.Data;
using http.Helper;
using Microsoft.AspNetCore.Mvc;
using Microsoft.EntityFrameworkCore;

namespace http.Endpoint.Kupac.GetRestorani
{
    [Route("Kupac-GetRestorani")]
    public class GetRestoraniEndpoint:MyBaseEndpoint<NoRequest, GetRestoraniResponse>
    {
        private readonly ApplicationDbContext _applicationDbContext;

        public GetRestoraniEndpoint(ApplicationDbContext applicationDbContext)
        {
            _applicationDbContext = applicationDbContext;
        }

        [HttpGet]
        public override async Task<GetRestoraniResponse> Akcija([FromQuery]NoRequest request, CancellationToken cancellationToken)
        {
            var restoran = await _applicationDbContext
                .Restoran.Where(x=>x.isObrisan==false)
                .Select(x => new GetRestoraniResponseRestorani
                {
                    ID=x.ID,
                    Naziv=x.Naziv,
                    Lokacija=x.Lokacija,
                    RadnoVrijemeDo=x.RadnoVrijemeDo,
                    RadnoVrijemeOd=x.RadnoVrijemeOd,
                    Opis=x.Opis,
                    SlikaRestorana=null,
                    ImePrezimeVlasnika=x.VlasnikRestorana.Ime+" "+x.VlasnikRestorana.Prezime
                }).ToListAsync();

            return new GetRestoraniResponse
            {
                Restorani = restoran,
                BrojRestorana=restoran.Count(),
            };


        }
    }
}
